_integration-tests/tests/net_http/serve_mux_handler.go

```diff
@@ -19,3 +20,14 @@ func (tc *TestCaseServeMuxHandler) Setup(t *testing.T) {
 	tc.handler = tc.serveMuxHandler()
 	tc.base.Setup(t)
 }
+
+type TestCaseHandlerIsNil struct {
+	base
+}
+
+func (tc *TestCaseHandlerIsNil) Setup(t *testing.T) {
+	http.HandleFunc("/hit", tc.base.handleHit)
+	http.HandleFunc("/", tc.base.handleRoot)
```

Comment:
This could introduce temporal coupling... I wonder if this should restore the default handler afterwards... But I'm also not entirely convinced this is actually any problem in this context... so just nothing this here for posterity in case it turns out it was and we stumble on this PR trying to figure out what is going on...